src/plugins/routing.js

```diff
@@ -12,148 +12,173 @@ import {
         !(strategy === STRATEGIES.NO_PREFIX) &&
         // no prefix for different domains
         !i18n.differentDomains
-      )
+    )
 
-      const path = (isPrefixed ? `/${locale}${route.path}` : route.path)
+    const path = (isPrefixed ? `/${locale}${route.path}` : route.path)
 
-      localizedRoute.path = path
-    } else {
-      // otherwise resolve route via the route name
-      // Build localized route options
-      let name = route.name + (strategy === STRATEGIES.NO_PREFIX ? '' : routesNameSeparator + locale)
+    localizedRoute.path = path
+  } else {
+    // otherwise resolve route via the route name
+    // Build localized route options
+    let name = route.name + (strategy === STRATEGIES.NO_PREFIX ? '' : routesNameSeparator + locale)
 
-      // Match route without prefix for default locale
-      if (locale === defaultLocale && strategy === STRATEGIES.PREFIX_AND_DEFAULT) {
-        name += routesNameSeparator + defaultLocaleRouteNameSuffix
-      }
+    // Match route without prefix for default locale
+    if (locale === defaultLocale && strategy === STRATEGIES.PREFIX_AND_DEFAULT) {
+      name += routesNameSeparator + defaultLocaleRouteNameSuffix
+    }
 
-      localizedRoute.name = name
+    localizedRoute.name = name
 
-      const { params } = localizedRoute
-      if (params && params['0'] === undefined && params.pathMatch) {
-        params['0'] = params.pathMatch
-      }
+    const { params } = localizedRoute
+    if (params && params['0'] === undefined && params.pathMatch) {
+      params['0'] = params.pathMatch
     }
-
-    // Resolve localized route
-    const router = this[routerPath]
-    const { route: { fullPath } } = router.resolve(localizedRoute)
-    return fullPath
   }
+
+  // Resolve localized route
+  const { route: { fullPath } } = this.router.resolve(localizedRoute)
+  return fullPath
 }
 
-function switchLocalePathFactory (i18nPath) {
-  return function switchLocalePath (locale) {
-    if (strategy === STRATEGIES.NO_PREFIX && locale && locale !== this[i18nPath].locale) {
-      // eslint-disable-next-line no-console
-      console.warn(`[${MODULE_NAME}] Passing non-current locale to switchLocalePath is unsupported when using no_prefix strategy`)
-    }
+function switchLocalePath (locale) {
+  const i18n = this.i18n
 
-    const name = this.getRouteBaseName()
-    if (!name) {
-      return ''
-    }
+  if (strategy === STRATEGIES.NO_PREFIX && locale && locale !== i18n.locale) {
+    // eslint-disable-next-line no-console
+    console.warn(`[${MODULE_NAME}] Passing non-current locale to switchLocalePath is unsupported when using no_prefix strategy`)
+  }
+
+  const name = this.getRouteBaseName()
+  if (!name) {
+    return ''
+  }
 
-    const { params, ...routeCopy } = this.$route
-    let langSwitchParams = {}
-    if (vuex && this.$store) {
-      langSwitchParams = this.$store.getters[`${vuex.moduleName}/localeRouteParams`](locale)
+  const route = this.route
```

Comment:
**nitpick:** Let's use object-destructuring here:

```suggestion
  const { route, store } = this
```

(and let's remove `const store = this.store` below)